docs/proposals/secrets.md

```diff
@@ -0,0 +1,80 @@
+
+
+### Details
+
+#### Broker Perspective:
+- Modify the Broker’s configuration file and add a new section for Secrets.
+- The Secrets section will contain a list of Secrets referred to by name which must exist in the Broker namespace.
+- Each referenced Secret will have a list of regexes for whitelisted APBs to apply the Secret to.
+    - For first iteration it will just be a 1-1 mapping of secret to apb name
+- It’s possible for an APB to be referenced in multiple secrets
+- If Broker is running an APB that matches an entry in the Secret config:
+    - Its pod is assigned ALL of the Secrets which match the configuration.
+    - The Secrets are mounted as files to a known location on the APB Pod.
+    - Secrets are mounted for all methods on the APB
+    - The APB runs in the namespace of the Secret.  (For first implementation this will be Broker namespace)
+    - The generated artifacts will continue to be populated in the namespace the user selects at time of provision, it’s only the ABP pod itself which will run in the namespace of the secret.
+    - During the /catalog operation the Broker will programmatically update the parameters associated with a ServiceClass/Plan.
+    - If the APB has an associated Secret AND the Secret exists, the Broker will read the ‘keys’ from the Secret and remove the matching parameters from being associated with the ServiceClass.
+    - This is done to remove the WebUI from prompting for this values which will obtained from Secret, yet it allows the APB author to continue to declare the parameters as they normally would, and in if no Secret is available the APB is still usable with those parameters entered.
+    - It’s possible for a Secret to not yet exist, the APB workflow will continue as is until the Secret is created.  The next invocation of the /catalog operation will update the parameters to reflect what Secrets exist at that point of time.
+
+
+#### APB Perspective
+Desire is to _not_ impact an APB author with this workflow.  APB author will declare the parameters required for the application to function as they normally would.  Details for obtaining the Secret info is handled in the base image.
+
+If the Secret exists for the APB, then the parameters will come from the Secret and the end user will not be prompted for them,  otherwise the APB will function as normal and it will be expected end user provides values for each parameter declared.
+
+#### APB Base Image
+Secrets will be mounted to a known location on the Pod
+Base Image will pass secret data in through the ansible mechanism for passing in extra variables in a file:
+“ -e @${PATH_TO_SECRETS}/my_secret_vars.yml”
+Ensuring no leakage of parameters contained in a secret through debug/log statements
+We will write a layer between the playbook and the output logs that removes any secrets from the log, guaranteeing that they will never show up in plain text
```

Comment:
yes, this will be a change to the base image, ideally a pure ansible solution, possibly a callback plugin or something similar. Broker shouldn't have to do anything for this.